internal/api/courses/assignments/submissions/submit_test.go

```diff
@@ -120,3 +123,53 @@ func TestRejectSubmissionMaxAttempts(test *testing.T) {
 			expected, responseContent.Message)
 	}
 }
+
+func TestRejectLateSubmission(test *testing.T) {
+	db.ResetForTesting()
+	defer db.ResetForTesting()
+
+	course := db.MustGetCourse("course101")
+	dueDate := timestamp.Zero()
+	course.Assignments["hw0"].DueDate = &dueDate
+	db.MustSaveCourse(course)
```

Comment:
This pattern is a bit strange (fetching/saving the course instead of the assignment).
I see that SaveAssignment() exists for DB backends, but is not exposed in internal/db like all the other functions.
Looks like an oversight to me.
I would just fix that and then use  db.MustGetTestAssignment() and db.SaveAssignment().